### PR TITLE
[Merged by Bors] - chore(MeasureTheory.Decomposition.Lebesgue): cleaning and a few new basic lemmas

### PR DESCRIPTION
**Pull request by RemyDegenne**
Move lemmas to put similar ones together, replace `refine'` by `refine` and `=>` by `↦`.
Lemmas added:
* `singularPart_add_rnDeriv` and `rnDeriv_add_singularPart`: almost aliases of `haveLebesgueDecomposition_add`
* `haveLebesgueDecomposition_smul'`, `haveLebesgueDecomposition_rnDeriv`
* `singularPart_eq_zero_of_ac`, `singularPart_eq_zero`, `singularPart_self`, `singularPart_eq_self`

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/MeasureTheory/Decomposition/Lebesgue.lean [hunk 1]
```diff
@@ -99,24 +91,64 @@ theorem haveLebesgueDecomposition_spec (μ ν : Measure α) [h : HaveLebesgueDec
...
+    μ.singularPart ν + ν.withDensity (μ.rnDeriv ν) = μ := (haveLebesgueDecomposition_add μ ν).symm
+
+lemma rnDeriv_add_singularPart (μ ν : Measure α) [HaveLebesgueDecomposition μ ν] :
+    ν.withDensity (μ.rnDeriv ν) + μ.singularPart ν = μ := by rw [add_comm, singularPart_add_rnDeriv]
+
+end ByDefinition
+
+section HaveLebesgueDecomposition
+
+instance instHaveLebesgueDecomposition_zero_left : HaveLebesgueDecomposition 0 ν where
```
Comment (sgouezel) [inline Mathlib/MeasureTheory/Decomposition/Lebesgue.lean:132]: instance names should not contain underscores, if I understand correctly. So
```suggestion
instance instHaveLebesgueDecompositionZeroLeft : HaveLebesgueDecomposition 0 ν where
```
Same thing below.

File: Mathlib/MeasureTheory/Decomposition/Lebesgue.lean [hunk 2]
```diff
@@ -228,18 +266,42 @@ lemma rnDeriv_zero (ν : Measure α) : (0 : Measure α).rnDeriv ν =ᵐ[ν] 0 :=
...
 
-lemma MutuallySingular.rnDeriv_ae_eq_zero {μ ν : Measure α} (hμν : μ ⟂ₘ ν) :
+lemma MutuallySingular.rnDeriv_ae_eq_zero (hμν : μ ⟂ₘ ν) :
     μ.rnDeriv ν =ᵐ[ν] 0 := by
   by_cases h : μ.HaveLebesgueDecomposition ν
   · rw [rnDeriv_eq_zero]
     exact hμν
   · rw [rnDeriv_of_not_haveLebesgueDecomposition h]
 
+theorem singularPart_withDensity (ν : Measure α) (f : α → ℝ≥0∞) :
```
Comment (sgouezel) [inline Mathlib/MeasureTheory/Decomposition/Lebesgue.lean:276]: make it a simp lemma?

File: Mathlib/MeasureTheory/Decomposition/Lebesgue.lean [hunk 3]
```diff
@@ -228,18 +266,42 @@ lemma rnDeriv_zero (ν : Measure α) : (0 : Measure α).rnDeriv ν =ᵐ[ν] 0 :=
...
+    (ν.withDensity f).singularPart ν = 0 :=
+  singularPart_eq_zero_of_ac (withDensity_absolutelyContinuous _ _)
+#align measure_theory.measure.singular_part_with_density MeasureTheory.Measure.singularPart_withDensity
+
 lemma rnDeriv_singularPart (μ ν : Measure α) :
     (μ.singularPart ν).rnDeriv ν =ᵐ[ν] 0 := by
   rw [rnDeriv_eq_zero]
   exact mutuallySingular_singularPart μ ν
 
+lemma singularPart_self (μ : Measure α) : μ.singularPart μ = 0 :=
```
Comment (sgouezel) [inline Mathlib/MeasureTheory/Decomposition/Lebesgue.lean:286]: simp lemma?

**Review by sgouezel — COMMENTED** [review]
bors d+
Thanks!

**Review by kex-y — APPROVED** [review]
Thanks! I'm happy with this

**Pull request closed without merging**